python/lsst/pipe/tasks/skyCorrection.py

```diff
@@ -387,6 +400,37 @@ def _restoreBackgroundRefineMask(self, calExps, calBkgs):
             skyCorrBases.append(skyCorrBase)
         return calExps, skyCorrBases
 
+    def _restoreVisitBackground(self, calExps, calBkgs):
+        """Restores the full focal-plane background to a visit.
+        Runs after _subtractSkyFrame() if doBgModel1=False.
+
+        Parameters
+        ----------
+        calExps : `list` [`lsst.afw.image.exposure.ExposureF`]
+            Calibrated exposures to be background subtracted.
+        calBkgs : `list` [`lsst.afw.math._backgroundList.BackgroundList`]
+            Background lists associated with the input calibrated exposures.
+
+        Returns
+        -------
+        calExps : `list` [`lsst.afw.image.maskedImage.MaskedImageF`]
+            Background subtracted exposures for creating a focal plane image.
+        calBkgs : `list` [`lsst.afw.math._backgroundList.BackgroundList`]
+            Updated background lists with a visit-level model removed.
+        """
+        for calExp, calBkg in zip(calExps, calBkgs):
+            image = calExp.getMaskedImage()
+
+            # Restore full focal-plane background in calexp; remove from BGList
+            skyCorrBase = calBkg[-2][0].getImageF()
+            image += skyCorrBase
+            calBkg._backgrounds.pop(-2)
+
+            self.log.info(
+                "Detector %d: FFP background restored",
```

Comment:
I think this log message could do with being more verbose, e.g.:
```
"Detector %d: Initial background model prior to sky frame subtraction (bgModel1) has been restored",
```

This does however now clash with usage of "initial background" on lines 377 and 396, which refer to the "original background" at the calexp level. I would favor renaming these two instances to "original background", and preserve initial background as a reference to `bgModel1`.